core/posts/serializers.py

```diff
@@ -1,15 +1,23 @@
 from rest_framework import serializers
 
 from core.posts.models import Posts, Comments
+from core.authors.serializers import AuthorSummarySerializer
 
 class PostsSerializer(serializers.ModelSerializer):
     visibleTo = serializers.ListField(child=serializers.CharField(max_length=100), default=list)
+    comments = serializers.PrimaryKeyRelatedField(many=True, read_only=True)
 
     class Meta:
         model = Posts
-        fields = ('id', 'post_id', 'author', 'source', 'origin', 'contentType', 'unlisted',
+        fields = ('id', 'post_id', 'author', 'source', 'origin', 'contentType', 'unlisted', 'comments',
             'content', 'title', 'description', 'published', 'visibleTo', 'visibility', "categories")
 
+    # Credits to Ivan Semochkin, https://stackoverflow.com/questions/41248271/django-rest-framework-not-responding-to-read-only-on-nested-data
+    def to_representation(self, instance):
+        representation = super(PostsSerializer, self).to_representation(instance)
+        representation['author'] = AuthorSummarySerializer(instance.author, read_only=True).data
```

Comment:
You can set instance.author.host to https://github.com/i-5ive/cmput404-project/blob/master/core/hostUtil.py#L7

and set instance.author.url to https://github.com/i-5ive/cmput404-project/blob/master/core/authors/util.py#L18

Reply:
god tier